File: Graphs/floyd warshall algorithm/floyd_warshall.js
function floydWarshall(graph) {
  const n = graph.length;
  const dist = new Array(n).fill().map(() => new Array(n).fill(Infinity));

  for (let i = 0; i < n; i++) {
    dist[i][i] = 0;
  }

  for (let u = 0; u < n; u++) {
    for (let v = 0; v < n; v++) {
      dist[u][v] = graph[u][v];
    }
  }

  for (let k = 0; k < n; k++) {
    for (let i = 0; i < n; i++) {
      for (let j = 0; j < n; j++) {
        if (dist[i][k] + dist[k][j] < dist[i][j]) {
          dist[i][j] = dist[i][k] + dist[k][j];
        }
      }
    }
  }

  return dist;
}

// Example usage:
const graph = [
  [0, 3, Infinity, 2],
  [Infinity, 0, Infinity, Infinity],
  [Infinity, 7, 0, 1],
  [6, Infinity, Infinity, 0],
];

const result = floydWarshall(graph);
for (const row of result) {
  console.log(row);
}

File: Graphs/floyd warshall algorithm/floyd_warshall.py
def floyd_warshall(graph):
    n = len(graph)
    dist = [[float('inf')] * n for _ in range(n)]

    for i in range(n):
        dist[i][i] = 0

    for u in range(n):
        for v in range(n):
            dist[u][v] = graph[u][v]

    for k in range(n):
        for i in range(n):
            for j in range(n):
                if dist[i][k] + dist[k][j] < dist[i][j]:
                    dist[i][j] = dist[i][k] + dist[k][j]

    return dist


# Example usage:
graph = [
    [0, 3, float('inf'), 2],
    [float('inf'), 0, float('inf'), float('inf')],
    [float('inf'), 7, 0, 1],
    [6, float('inf'), float('inf'), 0]
]

result = floyd_warshall(graph)
for row in result:
    print(row)
